### PR TITLE
ARROW-5773: [R] Clean up documentation before release

### PR DESCRIPTION
**Pull request by nealrichardson**
This cleans up a lot of the docs for the reader and writer functions, which are the main end-user-facing functions this release will expose. See a preview of the docs site at https://enpiar.com/arrow-r-site/reference/, with focus on the "above the fold" functions. 

There's a lot more to do, but I think this is the minimum before we submit to CRAN, and it would be great to get as much of the cleanup in 0.14 so that there's not a divergence between official 0.14 and CRAN 0.14.

This patch also adds a trivial "example" to one of the man pages, without which CRAN has been known to reject.

### CODE REVIEW

File: r/R/csv.R [hunk 1]
```diff
@@ -189,9 +189,10 @@ read_tsv_arrow <- function(file,
 `arrow::csv::ParseOptions` <- R6Class("arrow::csv::ParseOptions", inherit = `arrow::Object`)
 `arrow::csv::ConvertOptions` <- R6Class("arrow::csv::ConvertOptions", inherit = `arrow::Object`)
 
-#' read options for the csv reader
+#' Read options for the Arrow file readers
 #'
-#' @param block_size Block size we request from the IO layer; also determines the size of chunks when use_threads is `TRUE`
+#' @param use_threads Whether to use the global CPU thread pool
```
Comment (emkornfield) [inline r/R/csv.R]: should this be in the function declaration below?  I only see blocksize

File: r/R/csv.R [hunk 2]
```diff
@@ -289,20 +289,20 @@ csv_convert_options <- function(check_utf8 = TRUE) {
...
-#' @param parse_options, see [csv_parse_options()]
-#' @param convert_options, see [csv_convert_options()]
+#' @param read_options see [csv_read_options()]
+#' @param parse_options see [csv_parse_options()]
+#' @param convert_options see [csv_convert_options()]
 #' @param ... additional parameters.
 #'
-#' @return An `arrow::csv::TableReader` R6 object. Call `$Read()` on it to get
-#' an Arrow Table.
+#' @return An `arrow::csv::TableReader` or `arrow::json::TableReader` R6
```
Comment (emkornfield) [inline r/R/csv.R]: I only see what look like csv methods below?

**Comment by emkornfield** [comment]
@nealrichardson not familiar with R but a few of the documentation details looked strange?  I didn't look at what github classified as auto-generated files.

File: r/R/csv.R (hunk 1 above)
Comment (wesm) [reply r/R/csv.R]: I tried to fix this

File: r/R/csv.R (hunk 2 above)
Comment (wesm) [reply r/R/csv.R]: I agree this doesn't seem right

**Comment by wesm** [comment]
@nealrichardson seems there is one more issue, I can merge this once that's sorted out

File: r/R/csv.R (hunk 2 above)
Comment (nealrichardson) [reply r/R/csv.R]: `json_table_reader`'s docs are merged with `csv_table_reader` here: https://github.com/apache/arrow/pull/4736/files#diff-ae58975e47ac9d8718c62ee3c1807116R64

This is what the end result looks like: https://enpiar.com/arrow-r-site/reference/csv_table_reader.html

File: r/R/csv.R (hunk 1 above)
Comment (nealrichardson) [reply r/R/csv.R]: Thanks. This was a similar issue, where `json_read_options` was documented with `csv_read_options`, and the other function had the argument: https://enpiar.com/arrow-r-site/reference/csv_read_options.html

They should be the same so Wes's fix improves that.

**Comment by nealrichardson** [comment]
Responded; I think this is good to go.

**Comment by nealrichardson** [comment]
For future reference, https://travis-ci.org/apache/arrow/jobs/551636429#L3665 and that section of the `R CMD check` output would make noise if there were a mismatch between the function signatures and the generated docs, and the Travis build would fail.

**Comment by nealrichardson** [comment]
In fact, failure looks like this: https://travis-ci.org/nealrichardson/arrow/jobs/551782285#L3661-L3671

The man pages needed to be updated after Wes added the argument to `csv_read_options()` (you can run `make doc` in the `r/` directory to do that). Just did that and pushed.

**Comment by nealrichardson** [comment]
Passing travis build: https://travis-ci.org/nealrichardson/arrow/jobs/551796705

**Pull request closed without merging**